metrics-prometheus/src/main/java/com/alibaba/metrics/prometheus/DubboMetricsExports.java

```diff
@@ -0,0 +1,187 @@
+                metricFamilySamples.add(fromFastCompass(entry.getKey(), entry.getValue()));
+            }
+        }
+        return metricFamilySamples;
+    }
+
+    public MetricFamilySamples fromCounter(MetricName metricName, Counter counter) {
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "", new ArrayList<String>(), new ArrayList<String>(), new Long(counter.getCount()).doubleValue());
+        return new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(metricName.getKey(), counter), Arrays.asList(sample));
+    }
+
+    public MetricFamilySamples fromGauge(MetricName metricName, Gauge gauge) {
+        Object o = gauge.getValue();
+        double value;
+        if (o instanceof Number) {
+            value = ((Number) o).doubleValue();
+        } else if (o instanceof Boolean) {
+            value = ((Boolean) o) ? 1: 0;
+        } else {
+            return null;
+        }
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "", new ArrayList<String>(), new ArrayList<String>(), value);
+        return new MetricFamilySamples(sample.name, Type.GAUGE, getHelpMessage(metricName.getKey(), gauge), Arrays.asList
+                (sample));
+    }
+
+    public MetricFamilySamples fromTimer(MetricName metricName, Timer timer) {
+        return fromSnapshotAndCount(metricName, timer.getSnapshot(), timer.getCount(),
+                1.0D / TimeUnit.SECONDS.toNanos(1L), getHelpMessage(metricName.getKey(), timer));
+    }
+
+    public MetricFamilySamples fromMeter(MetricName metricName, Meter meter) {
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "_total", new ArrayList<String>(), new ArrayList<String>(), meter.getCount());
+        return new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(metricName.getKey(), meter), Arrays.asList
+                (sample));
+    }
+
+    public MetricFamilySamples fromHistogram(MetricName metricName, Histogram histogram) {
+        return fromSnapshotAndCount(metricName, histogram.getSnapshot(), histogram.getCount(), 1.0,
+                getHelpMessage(metricName.getKey(), histogram));
+    }
+
+    public MetricFamilySamples fromClusterHistogram(MetricName metricName, ClusterHistogram clusterHistogram) {
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        Map<Long, Map<Long, Long>> buckets = clusterHistogram.getBucketValues(0L);
+        for (Map.Entry<Long, Map<Long, Long>> entry : buckets.entrySet()) {
+            String suffix = "_time" + entry.getKey().toString();
+            Map<Long, Long> bucket = entry.getValue();
+            for (Map.Entry<Long, Long> entry1 : bucket.entrySet()) {
+                samples.add(sampleBuilder.createSample(metricName, suffix, Arrays.asList("bucket"), Arrays.asList(entry1.getKey().toString()), entry1.getValue()));
+            }
```

Comment:
There is one more bucket called `+Infinity`. see the test output for Prometheus Histogram.

```
# HELP test_prom_histogram test histogram
# TYPE test_prom_histogram histogram
test_prom_histogram_bucket{le="10.0",} 20.0
test_prom_histogram_bucket{le="100.0",} 172.0
test_prom_histogram_bucket{le="500.0",} 830.0
test_prom_histogram_bucket{le="1000.0",} 1615.0
test_prom_histogram_bucket{le="+Inf",} 1615.0
```

Note that in Prometheus the value is accumulated for each bucket, however in Alibaba metrics, the value is not. It may be necessary to adapt to prometheus logic.